core/data/core.joke

```diff
@@ -4189,3 +4196,190 @@
   {:added "1.0"}
   []
   (joker-version__))
+
+(defn- check-valid-options
+  "Throws an exception if the given option map contains keys not listed
+  as valid, else returns nil."
+  [options & valid-keys]
+  (when (seq (apply disj (apply hash-set (keys options)) valid-keys))
+    (throw
+      (ex-info ; was: (IllegalArgumentException.
+        (apply str "Only these options are valid: "
+          (first valid-keys)
+          (map #(str ", " %) (rest valid-keys))) {}))))
+
+;;multimethods
+(defn- space-str
```

Comment:
We should try to minimize the number of non-Clojure functions in the core namespace, even if they are private, to reduce the probability of name collisions. If necessary, such functions should be named with `__` suffix, so `space_str__` in this case. Although this function is only used in one place, so should probably be inlined. 
Also, the implementation could be simpler: `(apply str (interleave (repeat " ") more))`

Reply:
Nice! Hadn't used `interleave` before....